src/components/Fblogin.js

```diff
@@ -81,38 +133,65 @@ export default class Fblogin extends Component {
                 LoginManager.logOut();
                    }
 
-   shouldComponentUpdate(props, state) {
-        return state.getphone == true;
+
+   render() {
+        return (
+           <View style={{marginTop:20}}>
+            {this.state.getphone == true ?
+               [
+                 <TextInput
+                     placeholder = "enter your phone number"
+                     onChangeText = { (text) => this.setState({mobile_number : text})} />,
+                 <Button title="Submit my contact" onPress={this.sendUpdate} />
+               ]
+               :
+                [
+                  <Button title="Login with FaceBook" onPress={this.facebookLogin} />
+
+                 ]
+                 }
+
+         </View>
+      );
     }
 
-  render() {
-      return (
-         <View style={{marginTop:20}}>
-          {this.state.getphone == true ?
-             [
-               <TextInput
-                   placeholder = "enter your phone number"
-                   onChangeText = { (text) => this.setState({mobile_number : text})} />,
-               <Button title="Submit my contact" onPress={this.sendUpdate} />
-             ]
-             :
-              [
-                <Button title="Login with FaceBook" onPress={this.facebookLogin} />
+  getUpdateUser = async () => {
+         let userID = await AsyncStorage.getItem('user_id');
+         tokenvalue = await AsyncStorage.getItem('tokenval');
+         fcmtoken = await AsyncStorage.getItem('fcmToken');
+         UserService.getUserDetails(userID,tokenvalue).then(async (res) => {
+           let id = UserService.getUserId();
+
+           if (res) {
+             res.device_id = fcmtoken
+             return res;
+
+           } else {
+             throw new Error("Token updation failed")
+           }
+         }).then(async (res) => {
+            const checkUpdate = await UserService.UpdateUserDetails(res,tokenvalue);
+            if (checkUpdate == "success"){
+               this.props.navigation.navigate('Home');
+            } else {
+               throw new Error("Updating Token failed")
+              }
+         }).catch((error)=>{
+               alert(error);
+                    });
+
 
-               ]
-               }
 
-       </View>
-    );
-  }
+      }
+
 
 fbChecklogin = async () => {
     const {fbData} = this.state;
     let fb_id = fbTokenval.userID
     let fbtoken = fbTokenval.accessToken
 
   try{
-        let response = await fetch(baseUrl + 'FbUserLogin',{
+        let response = await fetch(baseUrl + 'FBUserLogin',{
```

Comment:
Duplicate call to FBUserLogin. Make a central call to the backend and use it wherever required.